tests/keras/preprocessing/image_test.py

```diff
@@ -53,10 +53,19 @@ def test_image_data_generator(self, tmpdir):
             generator.fit(images, augment=True)
 
             for x, y in generator.flow(images, np.arange(images.shape[0]),
-                                       shuffle=True, save_to_dir=str(tmpdir)):
-                assert x.shape[1:] == images.shape[1:]
+                                       shuffle=True, save_to_dir=str(tmpdir),
+                                       batch_size=3):
+                assert x.shape == images[:3].shape
                 break
 
+            # Test `flow` behavior as Sequence
+            seq = generator.flow(images, np.arange(images.shape[0]),
+                                 shuffle=True, save_to_dir=str(tmpdir),
```

Comment:
Need a test that shuffle works and on_epoch_end works

Reply:
Added tests for shuffling and reshuffling upon `on_epoch_end`